csrc/ir/internal_nodes.h

```diff
@@ -129,6 +129,48 @@ class IndexSelectOp : public Expr {
   }
 };
 
+class IndexPutAccumulateOp : public Expr {
+ public:
+  using Expr::Expr;
+
+  IndexPutAccumulateOp(
+      IrBuilderPasskey,
+      Val* out,
+      Val* acc,
+      Val* index,
+      Val* value);
+
+  NVFUSER_DECLARE_CLONE_AND_CREATE
+
+  const char* getOpString() const override {
+    return "IndexPutAccumulateOp";
+  }
+
+  std::string toString(int indent_size = 0) const override;
+  std::string toInlineString(int indent_size = 0) const override;
+  std::vector<PolymorphicValue> evaluate(
+      const ExpressionEvaluator& ee,
+      const std::vector<PolymorphicValue>& inputs) const override;
+
+  TensorView* accumulateTv() const {
+    return input(0)->as<TensorView>();
+  }
+
+  TensorView* indexTv() const {
+    return input(1)->as<TensorView>();
+  }
+
+  TensorView* valueTv() const {
+    return input(2)->as<TensorView>();
+  }
+
+  IterDomain* getIndexSeqID() const;
```

Comment:
What do you mean by "Seq"? Can you put some comments?

Reply:
That's the `sequence` ID. I added a note and a reference to the op signature in csrc/ops/indexing.cpp as well.

Reply:
This is a primitive IR node. I don't think it should use app-specific API names.

Reply:
You're adding a new primitive node to the nvFuser IR, and in the language of nvFuser IR, there's nothing called "sequence". Think of a case where this node is used for other applications where there may be nothing called sequence.

Reply:
I see, in this case, maybe it should have been

`getIndexedID` & `getIndexedIDValueTv` & `getConsumerOfIndexedID`

Reply:
Unsure. Please add enough comments.